model/registry/url.yaml

```diff
@@ -42,7 +42,11 @@ groups:
           In such case username and password SHOULD be redacted and attribute's value SHOULD be `https://REDACTED:REDACTED@www.example.com/`.
 
           `url.full` SHOULD capture the absolute URL when it is available (or can be reconstructed).
-          Sensitive content provided in `url.full` SHOULD be scrubbed when instrumentations can identify it.
+
+          Query string values SHOULD be redacted by default and replaced by the value `REDACTED`, e.g.
+          `https://www.example.com/path?abc=REDACTED&xyz=REDACTED` (the query string keys SHOULD be preserved).
+
+          Instrumentation MAY provide a configuration option to capture the full query string without any redaction.
```

Comment:
As this is registry definition, it is not specific to only http spans. 
From my understanding during last semconv meeting we have discussed that removing data by default is not the best option. And we should provide a user an option to get full url.

I see here 2 options:
- do this notation available only for specific use cases, such as http spans in this example
- we do redaction a default option with **SHOULD** notation only in case instrumentation has a switch to provide full url/query.

Reply:
Perhaps we want to apply the recommendation to redact to general-purpose instrumentations and leave a room for audit/security/other special telemetry to capture raw data?

I *think* the SHOULD in the above sentence already covers it and it should be ok if audit-related semconv would override  these notes. It should also be fine if specialized apps/distros enable raw data by default.

Reply:
> Perhaps we want to apply the recommendation to redact to general-purpose instrumentations and leave a room for audit/security/other special telemetry to capture raw data?

+1, in most cases, security requirements have higher priority than other requirements (e.g. privacy), for example, a privacy policy might say "do not collect client/user IP address" - which applies to general logging, while a security policy can say "client/user IP address must be collected for audit logs", and the latter wins.

Reply:
let me clarify it. I am concerned with word `MAY`. So we have here an option, where data is redacted (`SHOULD`), but instrumentation doesn't provide an option to get full content (`MAY`). I think in this case instrumentation should also have `SHOULD` notation

Reply:
```suggestion
          Instrumentation SHOULD provide a configuration option to capture the full query string without any redaction.
```